Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -38,56 +40,53 @@ ContourExtractor2DImageFilter<TInputImage>::ContourExtractor2DImageFilter()
 {
-  m_NumberOutputsAllocated = 0;
-  m_NumberOutputsWritten = 0;
-
   if (m_LabelContours) // each label has one or more contours
   {
     this->GenerateDataForLabels();
   }
   else // simple case of a single iso-value
   {
-    m_NumberLabelsRemaining = 1;
-    m_Interpolate = true;
     const InputRegionType region{ this->GetInput()->GetRequestedRegion() };
 
-    typename InputRegionType::SizeType shrunkSize = region.GetSize();
-    shrunkSize[0] -= 1;
-    shrunkSize[1] -= 1;
-    InputRegionType shrunkRegion(region.GetIndex(), shrunkSize);
-
-    this->CreateSingleContour(this->GetInput(),
-                              shrunkRegion,
-                              m_ContourValue,
-                              NumericTraits<InputPixelType>::max(),
-                              shrunkRegion.GetNumberOfPixels());
-  }
-
-  if (m_NumberOutputsWritten != m_NumberOutputsAllocated)
-  {
-    this->SetNumberOfIndexedOutputs(m_NumberOutputsWritten);
+    // Compute a shrunkRegion: we don't want the 3-by-3 SquareIterator
+    // to be centered in the bottom row or right column because then
+    // its lower-right 2-by-2 sub-square won't be wholly within the
+    // region.
+    const typename InputRegionType::SizeType shrunkSize{ region.GetSize()[0] - 1, region.GetSize()[1] - 1 };
+    const InputRegionType                    shrunkRegion(region.GetIndex(), shrunkSize);
+    // Since we are comparing to m_ContourValue, we don't care what label is supplied
+    // as the first argument of CreateSingleContour.
+    const InputPixelType        label{ NumericTraits<InputPixelType>::Zero };
+    std::vector<InputPixelType> allLabels;
+    allLabels.push_back(label);
+    std::unordered_map<InputPixelType, ContourContainerType> labelsContoursOutput;
+    labelsContoursOutput[label] = ContourContainerType{};
+    this->CreateSingleContour(label, shrunkRegion, shrunkRegion.GetNumberOfPixels(), labelsContoursOutput[label]);
+    FillOutputs(allLabels, labelsContoursOutput);
   }
 }
 
 
 template <typename TInputImage>
 void
-ContourExtractor2DImageFilter<TInputImage>::CreateSingleContour(const InputImageType * image,
-                                                                InputRegionType        shrunkRegion,
-                                                                InputRealType          lowerIsovalue,
-                                                                InputRealType          upperIsovalue,
-                                                                SizeValueType          totalNumberOfPixels)
+ContourExtractor2DImageFilter<TInputImage>::CreateSingleContour(InputPixelType         label,
+                                                                const InputRegionType  shrunkRegion,
+                                                                SizeValueType          totalNumberOfPixels,
+                                                                ContourContainerType & ContoursOutput)
```

Comment:
style: variable names should start with a lowercase letter

Reply:
Will do.